includes/class-internal-events.php

```diff
@@ -148,21 +148,20 @@ public function register_internal_events_schedules( $schedules ) {
 	 * Schedule internal events
 	 */
 	public function schedule_internal_events() {
-		$when = strtotime( sprintf( '+%d seconds', JOB_QUEUE_WINDOW_IN_SECONDS ) );
-
 		$schedules = wp_get_schedules();
 
 		foreach ( $this->internal_events as $event_args ) {
 			if ( ! wp_next_scheduled( $event_args['action'] ) ) {
-				$interval = array_key_exists( $event_args['schedule'], $schedules ) ? $schedules[ $event_args['schedule'] ]['interval'] : 0;
+				$interval  = array_key_exists( $event_args['schedule'], $schedules ) ? $schedules[ $event_args['schedule'] ]['interval'] : 0;
+				$timestamp = isset( $event_args['timestamp'] ) ? (int) $event_args['timestamp'] : strtotime( sprintf( '+%d seconds', JOB_QUEUE_WINDOW_IN_SECONDS ) );
```

Comment:
It seems to me that keeping this var `$when` makes it clearer its purpose (when, when_start, start, etc.)

Reply:
My thought process was to mirror the `wp_schedule_event()` naming: https://developer.wordpress.org/reference/functions/wp_schedule_event/. So those familiar with one could easily recognize the other.